src/enzyme_ad/jax/Dialect/Ops.cpp

```diff
@@ -356,14 +370,46 @@ class ReadNoneArg final : public OpRewritePattern<OpTy> {
   }
 };
 
+class RemoveUnusedJitCall final : public OpRewritePattern<JITCallOp> {
```

Comment:
In this specific case, I would recommend adding a folder: https://mlir.llvm.org/docs/Canonicalization/#canonicalizing-with-the-fold-method

Reply:
I am not sure how to remove the op with a folder. "Note that this mechanism cannot be used to remove 0-result operations."